walk/walk.go

```diff
@@ -111,8 +112,13 @@ func (f *File) AddReleaseFunc(fn ReleaseFunc) {
 // Stat checks if the file has changed by comparing its current state (size, mod time) to when it was first read.
 // It returns a boolean indicating if the file has changed, the current file info, and an error if any.
 func (f *File) Stat() (changed bool, info fs.FileInfo, err error) {
+	p := f.Path
+	if f.TmpPath != "" {
+		p = f.TmpPath
+	}
```

Comment:
It seems a bit strange to me that we have code floating around that stat-s these tmpfiles. It doesn't really make any sense to do.

Would it be better to crash here if `f.TmpPath` is nonempty, and instead update the callers to not call into this code?

Reply:
Stat before and after is used for the changed count in the stats we output at the end, indicating if any formatting resulted in an actual change. Less important for stdin, but I didn't see any reason to break the existing flow.